phpseclib/File/ASN1.php

```diff
@@ -849,12 +839,10 @@ public static function encodeDER($source, $mapping, $special = [])
      * ASN.1 Encode (Helper function)
      *
      * @param Element|string|array|null $source
-     * @param array $mapping
-     * @param int $idx
-     * @param array $special
+     * @param string|int|null $idx
      * @return string
      */
-    private static function encode_der($source, $mapping, $idx = null, $special = [])
+    private static function encode_der($source, array $mapping, $idx = null, array $special = [])
```

Comment:
I feel like `int $idx = null` should work here?

Reply:
Tests fail because a string is passed to `$idx`.